test/users.test.js

```diff
@@ -77,4 +79,52 @@ describe('Users', () => {
         .catch(err => done(err));
     });
   });
+
+  describe('POST /users/session', () => {
+    test('token should be returned', async done => {
+      const getUserByEmailSpy = jest.spyOn(UsersService, 'getUserByEmail');
+      getUserByEmailSpy.mockImplementation(getUserByEmailMock);
+      await request(app)
+        .post('/users/sessions')
+        .send(signInInput)
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .then(res => {
+          expect(res.body.data.token).toBeTruthy();
+          done();
+        })
+        .catch(err => done(err));
+    });
+
+    test('bad request should be returned when password does not match', async done => {
+      const getUserByEmailSpy = jest.spyOn(UsersService, 'getUserByEmail');
+      getUserByEmailSpy.mockImplementation(getUserByEmailMock);
+      signInInput.password = 'abc12345';
+      await request(app)
+        .post('/users/sessions')
+        .send(signInInput)
+        .expect('Content-Type', /json/)
+        .expect(400)
+        .then(res => {
+          expect(res.body.message).toBe(BAD_CREDENTIALS);
+          done();
+        })
+        .catch(err => done(err));
+    });
+
+    test('bad request should be returned when user does not exist', async done => {
+      const getUserByEmailSpy = jest.spyOn(UsersService, 'getUserByEmail');
+      getUserByEmailSpy.mockImplementation(getNullUserByEmailMock);
+      await request(app)
+        .post('/users/sessions')
+        .send(signInInput)
+        .expect('Content-Type', /json/)
+        .expect(400)
```

Comment:
same here. 400 indicates the request was malformed.  For the requests that the BDD could't find you should use 404, BUT this is a login request, so for security reasons, you must use 401 unauthorized.

Reply:
ok